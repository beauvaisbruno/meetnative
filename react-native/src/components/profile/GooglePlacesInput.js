"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const React = __importStar(require("react"));
// @ts-ignore
const react_native_google_places_autocomplete_1 = require("react-native-google-places-autocomplete");
const intl_1 = require("../../config/intl");
const lang_1 = __importDefault(require("./lang"));
const custom_1 = __importDefault(
  require("../../config/native-base-theme/variables/custom")
);
let persistCity = "";
exports.GooglePlacesInput = ({ city, onChange }) => {
  if (city && city !== "") persistCity = city;
  return React.createElement(
    react_native_google_places_autocomplete_1.GooglePlacesAutocomplete,
    {
      testID: "GooglePlacesAutocomplete",
      placeholder: lang_1.default("Search"),
      minLength: 2,
      autoFocus: false,
      returnKeyType: "search",
      listViewDisplayed: "false", // auto/true/false/undefined
      fetchDetails: true,
      enablePoweredByContainer: false,
      onPress: (data, details) => {
        // 'details' is provided when fetchDetails = true
        onChange(
          details.name,
          details.geometry.location.lat,
          details.geometry.location.lng
        );
      },
      getDefaultValue: () => persistCity,
      currentLocation: true,
      currentLocationLabel: lang_1.default("Current location"),
      query: {
        // available options: https://developers.google.com/places/web-service/autocomplete
        key: "hidden",
        language: intl_1.locale(),
        types: "(cities)"
      },
      nearbyPlacesAPI: "GooglePlacesSearch", // Which API to use: GoogleReverseGeocoding or GooglePlacesSearch
      GooglePlacesSearchQuery: {
        // available options for GooglePlacesSearch API : https://developers.google.com/places/web-service/search
        rankby: "distance",
        type: "locality"
      },
      debounce: 200,
      styles: {
        textInputContainer: {
          backgroundColor: "rgba(0,0,0,0)",
          borderTopWidth: 0,
          borderBottomWidth: 0
        },
        textInput: {
          marginTop: 4,
          marginLeft: 0,
          marginRight: 0,
          paddingLeft: 7,
          paddingRight: 0,
          height: 38,
          color: custom_1.default.inputColor,
          fontSize: 16
        },
        predefinedPlacesDescription: {
          color: custom_1.default.brandPrimary
        }
      }
    }
  );
};
