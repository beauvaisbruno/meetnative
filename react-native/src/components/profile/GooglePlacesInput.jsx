"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
// @ts-ignore
var react_native_google_places_autocomplete_1 = require("react-native-google-places-autocomplete");
var intl_1 = require("../../config/intl");
var lang_1 = require("./lang");
var custom_1 = require("../../config/native-base-theme/variables/custom");
var persistCity = "";
exports.GooglePlacesInput = function (_a) {
    var city = _a.city, onChange = _a.onChange;
    if (city && city !== "")
        persistCity = city;
    return (<react_native_google_places_autocomplete_1.GooglePlacesAutocomplete testID={"GooglePlacesAutocomplete"} placeholder={lang_1.default("Search")} minLength={2} autoFocus={false} returnKeyType={"search"} // https://facebook.github.io/react-native/docs/textinput.html#returnkeytype
     listViewDisplayed="false" // auto/true/false/undefined
     fetchDetails enablePoweredByContainer={false} onPress={function (data, details) {
        // 'details' is provided when fetchDetails = true
        onChange(details.name, details.geometry.location.lat, details.geometry.location.lng);
    }} getDefaultValue={function () { return persistCity; }} currentLocation currentLocationLabel={lang_1.default("Current location")} query={{
        // available options: https://developers.google.com/places/web-service/autocomplete
        key: "AIzaSyACed1wrMg8bwNMQTww_oPn8WQdkzkRM7I",
        language: intl_1.locale(),
        types: "(cities)"
    }} nearbyPlacesAPI="GooglePlacesSearch" // Which API to use: GoogleReverseGeocoding or GooglePlacesSearch
     GooglePlacesSearchQuery={{
        // available options for GooglePlacesSearch API : https://developers.google.com/places/web-service/search
        rankby: "distance",
        type: "locality"
    }} debounce={200} styles={{
        textInputContainer: {
            backgroundColor: "rgba(0,0,0,0)",
            borderTopWidth: 0,
            borderBottomWidth: 0
        },
        textInput: {
            marginTop: 4,
            marginLeft: 0,
            marginRight: 0,
            paddingLeft: 7,
            paddingRight: 0,
            height: 38,
            color: custom_1.default.inputColor,
            fontSize: 16
        },
        predefinedPlacesDescription: {
            color: custom_1.default.brandPrimary
        }
    }}/>);
};
