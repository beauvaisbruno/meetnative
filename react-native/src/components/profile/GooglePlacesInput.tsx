import * as React from "react";
// @ts-ignore
import { GooglePlacesAutocomplete } from "react-native-google-places-autocomplete";
import { locale } from "../../config/intl";
import lang from "./lang";
import custom from "../../config/native-base-theme/variables/custom";

let persistCity = "";

export const GooglePlacesInput = ({
  city,
  onChange
}: {
  city: string;
  onChange: (city: string, lat: number, lon: number) => void;
}) => {
  if (city && city !== "") persistCity = city;
  return (
    <GooglePlacesAutocomplete
      testID={"GooglePlacesAutocomplete"}
      placeholder={lang("Search")}
      minLength={2}
      autoFocus={false}
      returnKeyType={"search"} // https://facebook.github.io/react-native/docs/textinput.html#returnkeytype
      listViewDisplayed="false" // auto/true/false/undefined
      fetchDetails
      enablePoweredByContainer={false}
      onPress={(data: any, details: any) => {
        // 'details' is provided when fetchDetails = true
        onChange(
          details.name,
          details.geometry.location.lat,
          details.geometry.location.lng
        );
      }}
      getDefaultValue={() => persistCity}
      currentLocation
      currentLocationLabel={lang("Current location")}
      query={{
        // available options: https://developers.google.com/places/web-service/autocomplete
        key: "hidden",
        language: locale(),
        types: "(cities)"
      }}
      nearbyPlacesAPI="GooglePlacesSearch" // Which API to use: GoogleReverseGeocoding or GooglePlacesSearch
      GooglePlacesSearchQuery={{
        // available options for GooglePlacesSearch API : https://developers.google.com/places/web-service/search
        rankby: "distance",
        type: "locality"
      }}
      debounce={200}
      styles={{
        textInputContainer: {
          backgroundColor: "rgba(0,0,0,0)",
          borderTopWidth: 0,
          borderBottomWidth: 0
        },
        textInput: {
          marginTop: 4,
          marginLeft: 0,
          marginRight: 0,
          paddingLeft: 7,
          paddingRight: 0,
          height: 38,
          color: custom.inputColor,
          fontSize: 16
        },
        predefinedPlacesDescription: {
          color: custom.brandPrimary
        }
      }}
    />
  );
};
