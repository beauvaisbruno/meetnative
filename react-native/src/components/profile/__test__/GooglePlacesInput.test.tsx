import { shallow } from "enzyme";
import * as React from "react";
import { GooglePlacesInput } from "../GooglePlacesInput";
import toJson from "enzyme-to-json";

it("GooglePlacesInput", async () => {
  const onChange = jest.fn();
  const tree = shallow(
    <GooglePlacesInput onChange={onChange} city={"value"} />
  );
  expect(toJson(tree)).toMatchSnapshot();
  const GooglePlacesAutocomplete = tree.findWhere(
    node => node.prop("testID") === "GooglePlacesAutocomplete"
  );
  GooglePlacesAutocomplete.props().onPress(
    {},
    { name: "name", geometry: { location: { lat: 1, lng: 2 } } }
  );
  expect(onChange).toBeCalledWith("name", 1, 2);
});
