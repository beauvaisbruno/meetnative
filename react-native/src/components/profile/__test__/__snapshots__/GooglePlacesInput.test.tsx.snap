// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GooglePlacesInput 1`] = `
<GooglePlacesAutocomplete
  GooglePlacesSearchQuery={
    Object {
      "rankby": "distance",
      "type": "locality",
    }
  }
  GoogleReverseGeocodingQuery={Object {}}
  autoFillOnNotFound={false}
  autoFocus={false}
  currentLocation={true}
  currentLocationLabel="Current location"
  debounce={200}
  editable={true}
  enableEmptySections={true}
  enableHighAccuracyLocation={true}
  enablePoweredByContainer={false}
  fetchDetails={true}
  filterReverseGeocodingByTypes={Array []}
  getDefaultValue={[Function]}
  isRowScrollable={true}
  keyboardShouldPersistTaps="always"
  listViewDisplayed="false"
  minLength={2}
  nearbyPlacesAPI="GooglePlacesSearch"
  numberOfLines={1}
  onFail={[Function]}
  onNotFound={[Function]}
  onPress={[Function]}
  onSubmitEditing={[Function]}
  onTimeout={[Function]}
  placeholder="Search"
  placeholderTextColor="#A8A8A8"
  predefinedPlaces={Array []}
  predefinedPlacesAlwaysVisible={false}
  query={
    Object {
      "key": "AIzaSyACed1wrMg8bwNMQTww_oPn8WQdkzkRM7I",
      "language": "en",
      "types": "(cities)",
    }
  }
  returnKeyType="search"
  styles={
    Object {
      "predefinedPlacesDescription": Object {
        "color": "#8e7cc3",
      },
      "textInput": Object {
        "color": "#000",
        "fontSize": 16,
        "height": 38,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 4,
        "paddingLeft": 7,
        "paddingRight": 0,
      },
      "textInputContainer": Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
      },
    }
  }
  suppressDefaultStyles={false}
  testID="GooglePlacesAutocomplete"
  textInputHide={false}
  textInputProps={Object {}}
  timeout={20000}
  underlineColorAndroid="transparent"
/>
`;
